cli/cli/cmds_volume.go

```diff
@@ -339,7 +344,11 @@ func (c *CLI) initVolumeCmds() {
 			// volumes that are attached or can be attached should be valid
 			// candidates for a mount operation
 			result, err := c.lsVolumes(
-				args, volumeStatusAttached, volumeStatusAvailable)
+				args,
+				requestAttachments(
+					apitypes.VolumeAttachmentsMine,
+					apitypes.VolumeAttachmentsAttached,
+					apitypes.VolumeAttachmentsUnattached))
```

Comment:
@akutz.  I think this still has an issue. The block for `attachements.Mine` is filtering `vol.Attachments`, removing anything that does not have a matching IID completely. This makes sense on its own, but gives `Mine` precedence over `Attached` and `Unattached`.

For example,  the REX-Ray code that I commented on in this thread uses:
```
requestAttachments(
					apitypes.VolumeAttachmentsMine,
					apitypes.VolumeAttachmentsAttached,
					apitypes.VolumeAttachmentsUnattached))
```

The use of `Mine` with `Unattached` is still not going to get what it's asking for.

That being said, I'm still not sure that combination of masks makes sense. It essentially says "give me everything that's attached to me, everything that's attached at all (including any other hosts), and anything that is unattached". Doesn't that mean "give me everything"? Why use the bitmask at all, then, instead of just `Requested`?

Put another way, `Mine` still cannot be used with `attached` or `detached`, *if* the intent of `attached` and `detached` is to show you volumes that are not yours.

Using `Mine` + `Attached` + `Unattached` seems to have no purpose -- just use `Requested`
Using `Attached` + `Detached` in any scenario has no purpose -- at that point you get everything and should just use `Requested`.
Combining `Mine` with `Attached` seems misguided. `Mine` will always be a subset of `Attached`, and is thus pointless unless you submit `Mine` on its own.
Combining `Mine` with `Unattached` has potential. Basically, show me what I have and show what is available. But the current implementation would block that from working.

at least that's how I'm reading it.